### DISCUSSION TITLE
Running locally

### DISCUSSION THREAD

**Discussion by TechWiz-3** [guedesfelipe/pls-cli#41] [Q&A] [answered]
Hey man, great work with this project.

I have a Python CLI project with a similar setup and I'm wondering how you test changes locally since you're importing the package `pls_cli` instead of importing the local files?

Hope that makes sense!
Thanks

**Comment by guedesfelipe** [comment]
Thank you very much! Be welcome! 😃

I don't understand your question very well, can you elaborate more? 🤔

**Comment by TechWiz-3** [comment]
For example:

In `please.py` you import `from pls_cli.utils.quotes import get_rand_quote`

What happens if you want to test a change to `utils/quotes.py`? When `please.py` runs, it will import the package stored in `python/site-packages/pls-cli` rather than your local file `./utils/quotes`

My question is, when developing `pls-cli` do you have a special method to get around this problem?

Hope that's clearer 😅

**Comment by guedesfelipe** [comment, marked as answer]
If I understand the problem correctly, I use poetry to run locally ( And manage my environments ) when I'm developing, in this case it will get the local files and not the ones installed on my python machine.

I use that command when I'm developing:

```sh 
poetry run python please.py
```

**Comment by TechWiz-3** [comment]
Perfect, thanks a lot

**Reply by guedesfelipe** [reply]
If I helped you, you can mark as `Mark as answer`. And welcome to our Discuss 😃

**Reply by TechWiz-3** [reply]
Have you used `python3 setup.py develop` before? And if so how does it compare to poetry's `poetry run` functionality

**Reply by guedesfelipe** [reply]
I never used it this way

**Reply by TechWiz-3** [reply]
Thank you, you've helped me a lot 🙏

**Reply by guedesfelipe** [reply]
Welcome to our Discus! Feel free to ask any questions! 😃
